src/common/TransactionParser.ts

```diff
@@ -51,112 +47,139 @@ export class TransactionParser {
-        const newTransaction = transaction
-        newTransaction.gasUsed = receipt.gasUsed
+        const newTransaction = transaction;
+        newTransaction.gasUsed = receipt.gasUsed;
+        newTransaction.receipt = receipt;
         if (receipt.status) {
             newTransaction.error = receipt.status === "0x1" ? "" : "Error";
         }
-        return newTransaction
+        return newTransaction;
     }
 
     private extractTransactionData(block: any, transaction: any) {
-        const hash = String(transaction.hash);
         const from = String(transaction.from).toLowerCase();
         const to = String(transaction.to).toLowerCase();
-        const data: any =  {
-            _id: hash,
+
+        return {
+            _id: String(transaction.hash),
             blockNumber: Number(transaction.blockNumber),
             timeStamp: String(block.timestamp),
             nonce: Number(transaction.nonce),
-            from: from,
-            to: to,
+            from,
+            to,
             value: String(transaction.value),
             gas: String(transaction.gas),
             gasPrice: String(transaction.gasPrice),
             gasUsed: String(0),
             input: String(transaction.input),
             addresses: [from, to]
         };
-        return data
     }
 
     // ========================== OPERATION PARSING ========================== //
 
-    public parseTransactionOperations(transactions: any, contracts: any) {
-        if (!transactions || !contracts) {
-            return Promise.resolve();
-        }
+    public parseTransactionOperations(transactions: any[], contracts: any[]) {
+        if (!transactions || !contracts) return Promise.resolve();
 
         const operationPromises: any = [];
+
         transactions.map((transaction: any) => {
-            // find contract for this transaction
             const contract = contracts.find((c: any) => c.address === transaction.to);
-
-            if (contract) {
-                const decodedInput = erc20ABIDecoder.decodeMethod(transaction.input);
-                if (decodedInput && decodedInput.name === "transfer" && Array.isArray(decodedInput.params) && decodedInput.params.length == 2) {
-                    const to = decodedInput.params[0].value.toLowerCase();
-                    const value = removeScientificNotationFromNumbString(decodedInput.params[1].value);
-                    const p = this.findOrCreateTransactionOperation(transaction._id, transaction.from, to, value, contract._id);
-                    operationPromises.push(p);
+            const decodedLogs = erc20ABIDecoder.decodeLogs(transaction.receipt.logs);
+                // Covers where contract https://etherscan.io/tx/0x83388873b5a6d2d462e1f61e09dfcc6df779e2da46c8a1727f76e0e68b4c5ecf
+                if (contract) {
+                    if (decodedLogs.length > 0) {
+                        console.log("decodedLogs", decodedLogs)
```

Comment:
Remove console.log and other comments here.

Reply:
Fixed